hazelcast-spring/src/main/resources/hazelcast-spring-5.1.xsd

```diff
@@ -4637,6 +4639,143 @@
         </xs:attribute>
     </xs:complexType>
 
+    <xs:complexType name="device">
+        <xs:annotation>
+            <xs:documentation>
+                Device configuration.
+            </xs:documentation>
+        </xs:annotation>
+        <xs:all>
+            <xs:element name="base-dir" type="xs:string" minOccurs="0" default="tiered-store">
+                <xs:annotation>
+                    <xs:documentation>
+                        Device base directory that will be used for tiered-store.
+                    </xs:documentation>
+                </xs:annotation>
+            </xs:element>
+            <xs:element name="block-size" minOccurs="0" default="4096">
+                <xs:annotation>
+                    <xs:documentation>
+                        Device block/sector size in bytes. The default is 4096. The minimum is 512.
+                    </xs:documentation>
+                </xs:annotation>
+                <xs:simpleType>
+                    <xs:restriction base="xs:unsignedInt">
+                        <xs:minInclusive value="512"/>
```

Comment:
Is there any advantage of limiting this at the configuration level? There might be new devices in the future that has smaller block size. For example, Optane can be byte-addressed (in AppDirect mode), and the device's internal block size is 256 bytes.

Reply:
Yeah, the idea was to prevent users from (potentially) setting block size to some small value (1 for example), which would hurt performance and might lead to some weird behavior(s).

Reply:
I think we can relax it in the future, if a need arises, wdyt?

Reply:
Thanks, let's keep it as is for now. If we intend to support Optane as a device (in AppDirect mode), we'll need more changes than just tuning the block size anyways.